src/main/java/hudson/plugins/jira/JiraRestService.java

```diff
@@ -238,30 +245,41 @@ public BasicIssue createIssue(String projectKey, String description, String assi
 
     public User getUser(String username) {
         try {
-            return jiraRestClient.getUserClient().getUser(username).get(10, TimeUnit.SECONDS);
+            return jiraRestClient.getUserClient().getUser(username).get(timeout, TimeUnit.SECONDS);
         } catch (Exception e) {
             LOGGER.warning("jira rest client get user error. cause: " + e.getMessage());
             return null;
         }
     }
-
+ 
     public void updateIssue(String issueKey, List<Version> fixVersions) {
         final IssueInput issueInput = new IssueInputBuilder().setFixVersions(fixVersions)
                                                              .build();
         try {
-            jiraRestClient.getIssueClient().updateIssue(issueKey, issueInput).get(10, TimeUnit.SECONDS);
+            jiraRestClient.getIssueClient().updateIssue(issueKey, issueInput).get(timeout, TimeUnit.SECONDS);
         } catch (Exception e) {
             LOGGER.warning("jira rest client update issue error. cause: " + e.getMessage());
         }
     }
-
+    
+    public void setIssueLabels(String issueKey, List<String> labels) {
+        final IssueInput issueInput = new IssueInputBuilder()
+        		.setFieldValue(IssueFieldId.LABELS_FIELD.id, labels)
+                .build();
+        try {
+            jiraRestClient.getIssueClient().updateIssue(issueKey, issueInput).get(timeout, TimeUnit.SECONDS);
+        } catch (Exception e) {
+            LOGGER.log(Level.WARNING, "jira rest client update labels error for issue "+issueKey, e);
```

Comment:
Yes is, personally i always prefer slf4j, but i don't think it be happily to add two different loggers in one class or to change logging in whole class in this pull request (unrelated changes - maybe change logging in whole project later if someone have time to do).